rust/src/exchange/cause.rs

```diff
@@ -0,0 +1,51 @@
+// Software Name: its-client
+// SPDX-FileCopyrightText: Copyright (c) 2016-2023 Orange
+// SPDX-License-Identifier: MIT License
+//
+// This software is distributed under the MIT license, see LICENSE.txt file for more details.
+//
+// Author: Nicolas BUFFON <nicolas.buffon@orange.com> et al.
+// Software description: This Intelligent Transportation Systems (ITS) [MQTT](https://mqtt.org/) client based on the [JSon](https://www.json.org) [ETSI](https://www.etsi.org/committee/its) specification transcription provides a ready to connect project for the mobility (connected and autonomous vehicles, road side units, vulnerable road users,...).
+
+use crate::exchange::message::Message;
+use crate::exchange::Exchange;
+use std::fmt::Formatter;
+
+/// Cause is hereby used pass information to the [monitoring function][1] about which message a DENM
+/// was emitted from
+///
+/// If the `analyze(...)` call of an Analyzer implementation return some DENM to send, the info in
+/// the message that as provided to the method will be used to build a Caused that will allow to
```

Comment:
Perhaps typing errors here: should it be :
the message that has been provided....build a Cause that...
instead of:
the message that as provided...build a Caused that...